src/test/java/core/basesyntax/service/impl/ReaderServiceImplTest.java

```diff
@@ -0,0 +1,54 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.basesyntax.service.ReaderService;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class ReaderServiceImplTest {
+
+    private static ReaderService readerService;
+    private static List<String> expected;
+
+    @BeforeAll
+    public static void init() {
+        readerService = new ReaderServiceImpl();
+    }
+
+    @Test
+    void reader_fileNotExist_notOk() {
+        expected = new ArrayList<>();
+        String nonExistingFile = "not_exist";
+        assertThrows(RuntimeException.class,
+                () -> readerService.readFromFile(nonExistingFile),
+                "Should throw runtime exception.");
+    }
+
+    @Test
+    void reader_fileIsEmpty_notOk() {
+        expected = new ArrayList<>();
+        expected = readerService.readFromFile("src/main/resources/input_file.csv");
+        assertThrows(AssertionError.class,
+                () -> assertTrue(expected.isEmpty()),
+                "Should throw runtime exception.");
+    }
+
+    @Test
+    void reader_readFromFile_ok() {
```

Comment:
please read about our naming convention for test methods and fix this problem in all places: `<methodUnderTest>_<state>_<expectedBehavior> `
```suggestion
    void readFromFile_validInput_ok() {
```
'validInput' could be replaced with the actual conditions/state. Please fix the same issue in all places